src/compile/legend/assemble.ts

```diff
@@ -26,13 +26,18 @@ export function assembleLegends(model: Model): VgLegend[] {
 
   return flatten(vals(legendByDomain)).map((legendCmpt: LegendComponent) => {
     const legend = legendCmpt.combine();
-    // For non color channel's legend, we need to override symbol stroke config from Vega config
+
     if (legend.encode && legend.encode.symbols) {
       const out = legend.encode.symbols.update;
       if (out.fill && out.fill['value'] !== 'transparent' && !out.stroke && !legend.stroke) {
         // For non color channel's legend, we need to override symbol stroke config from Vega config if stroke channel is not used.
         out.stroke = {value: 'transparent'};
       }
+
+      if (legend.fill) {
```

Comment:
This solves the problem, but I'm wonder if fill shouldn't even be in the component in the first place. (Should we fix this in the parse stage, not here?)

Reply:
I think the issue is only with multiple channels sharing the same domain, and merging these occurs in this stage.